superset/common/query_object.py

```diff
@@ -236,7 +237,7 @@ def __init__(
     @property
     def metric_names(self) -> List[str]:
         """Return metrics names (labels), coerce adhoc metrics to strings."""
-        return get_metric_names(self.metrics)
+        return get_metric_names(self.metrics or [])
```

Comment:
I'm almost wondering if we should be returning `None` here if `self.metrics is None`

Reply:
I'd rather keep the prevalence of `None` minimum. `metric_names` is just a shortcut for presentational information so it should be OK to be less strict.